src/Helper/server-requests.js

```diff
@@ -0,0 +1,47 @@
+import axios from "axios";
+import toast from "react-hot-toast";
+
+export const getVideos = async (setIsLoading, setVideos) => {
+  try {
+    setIsLoading(true);
+    const res = await axios.get("/api/videos");
+    setVideos(res.data.videos);
+  } catch (error) {
+    console.error(error);
+  } finally {
+    setIsLoading(false);
+  }
+};
+
+export const login = async (
+  email,
+  password,
+  dispatch,
+  navigate,
+  toPreviousRoute
+) => {
+  try {
+    const res = await axios.post("/api/auth/login", { email, password });
+    dispatch({ type: "LOGIN", payload: res.data });
+    navigate(toPreviousRoute);
+    toast.success("Logged in successfully!");
+  } catch (error) {
+    if (error.response.status === 401 || error.response.status === 404) {
+      toast.error(error.response.data.errors[0]);
+    } else {
+      toast.error(error.message);
+    }
+    console.error(error);
```

Comment:
this could be removed b'coz you implemented toast

Reply:
I used it so that the user may have detailed info about the error if they want.